docs/api/api.yml

```diff
@@ -0,0 +1,89 @@
+openapi: 3.0.0
+info:
+  title: Download Service API
+  version: 1.1.2
+  description: |
+    This document specifies the Download Service API. It provides a direct interface to download files from the archive.
+    This API implements the operations `GET` on the paths `/` and `/ark:/:NAAN/:uuid/*`.
+    The NAAN represents the configured 'name assigning authority number' which for DANS is 73189.
+    It is not neccessary for this service to supply the bag store the bag is stored in.
+
+  license:
+    name: Apache 2.0
+    url: http://www.apache.org/licenses/LICENSE-2.0
+
+servers:
+  - url: 'http://localhost:20160'
+  - url: 'http://test.dans.knaw.nl:20160'
+
+paths:
+  /:
+    get:
+      summary: Returns a description of the service.
+      description:
+        This is the starting place for navigating the Download Service. It returns a description of the service.
+      responses:
+        200:
+          description: Description of the service
+          content:
+            text/plain:
+              schema:
+                type: string
+                description: A plain text, human readable description of the service
+                example: EASY Download Service running v1.1.2.
+        500:
+          $ref: '#/components/responses/Unavailable'
+
+  /ark:/{NAAN}/{ItemId}:
+    get:
+      summary: Returns an output stream of the file specified by the ItemId.
+      description: |
+        An item is either a directory or file in a bag. The item which is fetched is specified by the item-id, which is the bag-id followed by the local path of the file in the bag.
+        This service requires the ItemId to refer to a payload file, it is unable to download directories:
+
+        ```
+        example valid ItemId:
+         40594b6d-8378-4260-b96b-13b57beadf7c/path/to/file
+        ```
+
+      parameters:
+        - $ref: '#/components/parameters/ItemId'
+        - $ref: '#/components/parameters/NAAN'
+      responses:
+        200:
+          description: The file stream will be returned
+        400:
```

Comment:
I wonder when a 400 is returned and when a 404. This has been a source of confusion in the bag-store API itself as well. Does an "invalid bag-id" mean a string that could never be a bag-id, because it is not a UUID, or also a well-formed UUID which does not refer to an existing bag?

Reply:
In theory this should be the distinction yes, 400 should be returned in case of a malformed UUID, while a 404 should be returned in the case of a non-existent valid bag-id. However the uuid parser still seems to give inconsistent results when it comes to this. 

For instance let's assume the example UUID: `12341234-1234-1234-1234-123412341234` is the only bag-id in a bag-store. 
If any other valid UUID is used then the service will return a 404 as expected.
In the case of a UUID `12341234-1234-1234-1234-12341234123` however a 400 response seems more appropriate, but a 404 is still given.
When trying to take this to the extreme, in the case of a uuid: `0-1-2-3-4` the service still returns a 404 status code with the message: "not found: 00000000-0001-0002-0003-000000000004/data/a/deeper/path/some file.txt" which confirms that the parser will consider the uuid valid as long as it consists of the correct number of `-` separated segments.

In the case of the UUID containing more digits per segment than would be expected the service also returns a 404 instead of a 400.
For example the UUID `12341234-1234-1234-1234-1234123412341` would be parsed as `12341234-1234-1234-1234-234123412341` and would be considered valid.

Reply:
In easy-bag-store `nl.knaw.dans.easy.bagstore/ItemId.scala` implements the method `validateUuidLength` which checks that a UUID contains exactly 36 characters. If it does not it will return an IllegalArgumentException.

Reply:
To avoid confusion, I think it is better to limit 400 to an invalid request *body* and to return 404 for any URL that cannot be resolved (possibly because it contains a malformed UUID).